contracts/DomainRegistryV2.sol

```diff
@@ -101,27 +170,28 @@ contract DomainRegistryV2 is Initializable, OwnableUpgradeable {
         if (parentDomainOwner == address(0)) {
             parentReward = 0;
         } else {
-            parentReward = price * REWARD_PERCENT_OWNER / 100;
+            parentReward = (price * REWARD_PERCENT_OWNER) / 100;
         }
         uint256 ownerReward = price - parentReward;
 
         domains[rootDomain] = msg.sender;
         shares[msg.sender] += ownerReward;
```

Comment:
Wow, and how can the contract owner get its ether in the `withdraw()` function if it is being written not to its address?? Check the logic here and in the withdraw() attentively and let me know if I'm missing something. Do tests from different addresses as it would be in reality, if you didn't already
   -2 points for now

Reply:
agree its a bit confusing, it was supposed that all shares are going to owner, but in this case we could use only uint total value. Now I updated it in https://github.com/vpmoon/blockchain-playground/pull/8/files